test/GPUActivityAgentTest.cpp

```diff
@@ -0,0 +1,434 @@
+    // setup & load policy
+    policy[FREQ_MAX] = m_freq_max;
+    policy[FREQ_EFFICIENT] = m_freq_max / 2;
+    policy[PHI] = 0.1;
+    m_agent->validate_policy(policy);
+
+    // validate policy is modified as expected
+    // as phi --> 0 FREQ_EFFICIENT --> FREQ_MAX
+    ASSERT_EQ(m_num_policy, policy.size());
+    EXPECT_EQ(m_freq_max, policy[FREQ_MAX]);
+    EXPECT_GE(policy[FREQ_EFFICIENT], m_freq_max / 2);
+    EXPECT_LE(policy[FREQ_EFFICIENT], m_freq_max);
+    EXPECT_EQ(0.1, policy[PHI]);
+
+    //Fe > Fmax --> Error
+    policy[FREQ_MAX] = NAN;
+    policy[FREQ_EFFICIENT] = m_freq_max + 1;
+    policy[PHI] = NAN;
+    GEOPM_EXPECT_THROW_MESSAGE(m_agent->validate_policy(policy), GEOPM_ERROR_INVALID,
+                              "GPU_FREQ_EFFICIENT out of range");
+
+    //Fe < Fmin --> Error
+    policy[FREQ_MAX] = NAN;
+    policy[FREQ_EFFICIENT] = m_freq_min - 1;
+    policy[PHI] = NAN;
+    GEOPM_EXPECT_THROW_MESSAGE(m_agent->validate_policy(policy), GEOPM_ERROR_INVALID,
+                               "GPU_FREQ_EFFICIENT out of range");
+
+    //Fe > Policy Fmax --> Error
+    policy[FREQ_MAX] = m_freq_max - 2;
+    policy[FREQ_EFFICIENT] = m_freq_max - 1;
+    policy[PHI] = NAN;
+    GEOPM_EXPECT_THROW_MESSAGE(m_agent->validate_policy(policy), GEOPM_ERROR_INVALID,
+                               "value exceeds GPU_FREQ_MAX");
+
+    //Policy Fmax > Fmax --> Error
+    policy[FREQ_MAX] = m_freq_max + 1;
+    policy[FREQ_EFFICIENT] = NAN;
+    policy[PHI] = NAN;
+    GEOPM_EXPECT_THROW_MESSAGE(m_agent->validate_policy(policy), GEOPM_ERROR_INVALID,
+                               "GPU_FREQ_MAX out of range");
+
+    //Policy Fmax < Fmin --> Error
+    policy[FREQ_MAX] = m_freq_min - 1;
+    policy[FREQ_EFFICIENT] = NAN;
+    policy[PHI] = NAN;
+    GEOPM_EXPECT_THROW_MESSAGE(m_agent->validate_policy(policy), GEOPM_ERROR_INVALID,
+                               "GPU_FREQ_MAX out of range");
+
+    //Policy Phi < 0 --> Error
+    policy[FREQ_MAX] = NAN;
+    policy[FREQ_EFFICIENT] = NAN;
+    policy[PHI] = -1;
+    GEOPM_EXPECT_THROW_MESSAGE(m_agent->validate_policy(policy), GEOPM_ERROR_INVALID,
+                               "POLICY_GPU_PHI value out of range");
+
+    //Policy Phi > 1.0 --> Error
+    policy[FREQ_MAX] = NAN;
+    policy[FREQ_EFFICIENT] = NAN;
+    policy[PHI] = 1.1;
+    GEOPM_EXPECT_THROW_MESSAGE(m_agent->validate_policy(policy), GEOPM_ERROR_INVALID,
+                               "POLICY_GPU_PHI value out of range");
+}
+
+TEST_F(GPUActivityAgentTest, adjust_platform_high)
+{
+    //TODO: Setup f_max, min etc...via read
+    EXPECT_CALL(*m_platform_io, read_signal("GPU_CORE_FREQUENCY_MIN_AVAIL", _, _)).WillRepeatedly(
+                Return(m_freq_min));
+    EXPECT_CALL(*m_platform_io, read_signal("GPU_CORE_FREQUENCY_MAX_AVAIL", _, _)).WillRepeatedly(
+                Return(m_freq_max));
+    EXPECT_CALL(*m_platform_topo, num_domain(GEOPM_DOMAIN_BOARD)).WillRepeatedly(Return(M_NUM_BOARD));
```

Comment:
This block of code is repeated at the beginning of every test, so it could be moved into the ``SetUp()`` method.